data/helpers.d/utils

```diff
@@ -685,3 +699,37 @@ ynh_compare_current_package_version() {
     # Return the return value of dpkg --compare-versions
     dpkg --compare-versions $current_version $comparison $version
 }
+
+# Check if we should enforce sane default permissions (= disable rwx for 'others')
+# on file/folders handled with ynh_setup_source and ynh_add_config
+#
+# [internal]
+#
+# Having a file others-readable or a folder others-executable(=enterable)
+# is a security risk comparable to "chmod 777"
+#
+# Configuration files may contain secrets. Or even just being able to enter a
+# folder may allow an attacker to do nasty stuff (maybe a file or subfolder has
+# some write permission enabled for 'other' and the attacker may edit the
+# content or create files as leverage for priviledge escalation ...)
+#
+# The sane default should be to set ownership to $app:$app.
+# In specific case, you may want to set the ownership to $app:www-data
+# for example if nginx needs access to static files.
+#
+_ynh_apply_default_permissions() {
+    local target=$1
+
+    local ynh_requirement=$(jq -r '.requirements.yunohost' $YNH_APP_BASEDIR/manifest.json)
+
+    if [ -z "$ynh_requirements" ] || [ "$ynh_requirements" == "null" ] || dpkg --compare-versions $ynh_requirements ge 4.2
+    then
+        chmod o-rwx $target
```

Comment:
```suggestion
        chmod -R 440 $target
```

Reply:
I think o-rwx is likely best here, 440 would add a lot of additional work for package maintainers with no real benefit in terms of security

Reply:
Because you would need to find every single executable and +x that individually. Keeping the executable flags of the source tarball is a lot more reasonable

Reply:
Maybe `o-rwx` `g-w`?

Reply:
Agreed for `g-w`